File: backend/app/Models/Slideshow.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Slideshow extends Model
{
    protected $fillable = [
        'name',
        'description',
        'price',
        'image',
        'product_id',
        'enable',
        'link',
        'ssorder'
    ];

    protected $casts = [
        'enable' => 'boolean',
        'price' => 'decimal:2',
    ];

    /**
     * Get the product associated with the slideshow.
     */
    public function product()
    {
        return $this->belongsTo(Product::class);
    }
}

File: frontend/lib/models/slideshow.dart
class Slideshow {
  final int id;
  final String name;
  final String description;
  final double price;
  final String image;
  final int productId;
  final bool enable;
  final String? link;
  final int? ssorder;


  Slideshow({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.image,
    required this.productId,
    this.enable = true,
    this.link,
    this.ssorder,
  });
}